graphql_compiler/__init__.py

```diff
@@ -198,3 +199,74 @@ def get_graphql_schema_from_orientdb_schema_data(schema_data, class_to_field_typ
+            - query: string, the resulting compiled and parameterized query string
+            - language: string, specifying the language to which the query was compiled
+            - output_metadata: dict, output name -> OutputMetadata namedtuple object
+            - input_metadata: dict, name of input variables -> inferred GraphQL type, based on use
+    """
+    compilation_result = compile_graphql_to_cypher(
+        schema, graphql_query, type_equivalence_hints=type_equivalence_hints)
+    return compilation_result._replace(
+        query=insert_arguments_into_query(compilation_result, parameters))
+
+
+def graphql_to_cypher_neo4j(schema, graphql_query, parameters, type_equivalence_hints=None):
+    """Compile the GraphQL input using the schema into a Cypher query and associated metadata.
+
+    Args:
+        schema: GraphQL schema object describing the schema of the graph to be queried
+        graphql_query: the GraphQL query to compile to Cypher, as a string
+        parameters: dict, mapping argument name to its value, for every parameter the query expects.
+        type_equivalence_hints: optional dict of GraphQL interface or type -> GraphQL union.
+                                Used as a workaround for GraphQL's lack of support for
+                                inheritance across "types" (i.e. non-interfaces), as well as a
+                                workaround for Gremlin's total lack of inheritance-awareness.
+                                The key-value pairs in the dict specify that the "key" type
+                                is equivalent to the "value" type, i.e. that the GraphQL type or
+                                interface in the key is the most-derived common supertype
+                                of every GraphQL type in the "value" GraphQL union.
+                                Recursive expansion of type equivalence hints is not performed,
+                                and only type-level correctness of this argument is enforced.
+                                See README.md for more details on everything this parameter does.
+                                *****
+                                Be very careful with this option, as bad input here will
+                                lead to incorrect output queries being generated.
+                                *****
+
+    Returns:
+        a CompilationResult object, containing:
+            - query: string, the resulting compiled and parameterized query string
+            - language: string, specifying the language to which the query was compiled
+            - output_metadata: dict, output name -> OutputMetadata namedtuple object
+            - input_metadata: dict, name of input variables -> inferred GraphQL type, based on use
+    """
+    compilation_result = compile_graphql_to_cypher(
+        schema, graphql_query, type_equivalence_hints=type_equivalence_hints)
+    return compilation_result
```

Comment:
I'm not thrilled with this approach, because all the other methods have this line instead:

```    
return compilation_result._replace(
        query=insert_arguments_into_query(compilation_result, parameters))
```
But this one doesn't because in `query_formatting/common.py`, line 136, we call `insert_arguments_into_cypher_query_redisgraph` which is RedisGraph specific (since RedisGraph doesn't support query parameters at all). The condition checks that the language is Cypher, so if we called `insert_arguments_into_query` here then we'd essentially be manually interpolating the arguments in.

Open to discussing alternative ways to organize this code.

Reply:
Might be best to discuss this together in front of a whiteboard, it's kind of an unfortunate situation.